crypto/src/merkle/mod.rs

```diff
@@ -6,8 +6,17 @@
 use crate::{errors::MerkleTreeError, hash::Hasher};
 use core::slice;
 use math::log2;
+
+#[cfg(feature = "alloc")]
+use alloc::collections::BTreeSet;
+#[cfg(feature = "alloc")]
+use hashbrown::HashMap;
+#[cfg(feature = "std")]
 use std::collections::{BTreeSet, HashMap};
```

Comment:
If we have `utils::collections` module as described in the other comment, we could replace this with use `utils::collections::*`;